File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/adapter/transport/Bike.java
// Bike.java

package com.bigboxcode.designpattern.adapter.transport;

public class Bike implements Transport {
    @Override
    public int getNumberOfWheels() {
        return 2;
    }

    @Override
    public double getWeight() {
        return 700;
    }

    @Override
    public double getDistanceTravelled() {
        return 80;
    }

    @Override
    public double getTravelCostPerMile() {
        return 4;
    }
}

File: design-patterns/php/adapter/transportadapter/Bike.php
<?php
// Bike.php

namespace BigBoxCode\DesignPattern\Adapter\TransportAdapter;


class Bike implements Transport {
    public function getNumberOfWheels(): int {
        return 2;
    }

    public function getWeight(): float {
        return 700;
    }

    public function getDistanceTravelled(): float {
        return 80;
    }

    public function getTravelCostPerMile(): float {
        return 4;
    }
}